### PR TITLE
[IMP] mail: Disable escape on composer wizard

### PR DESCRIPTION
**Pull request by maki-odoo**
Disable escape on composer wizard

task #2793003
- rename Cancel to Discard on composer wizard
- disable escape on composer wizard to prevent closing the composer accidentally

### CODE REVIEW

**Review by alexkuhn — CHANGES_REQUESTED** [review]

**Comment by maki-odoo** [comment]
WIP

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

**Review by alexkuhn — APPROVED** [review]
LGTM for the `mail/` part

File: addons/mail/wizard/mail_compose_message_views.xml [hunk 1]
```diff
@@ -79,7 +79,7 @@
                         <button string="Log" name="action_send_mail"
                                 type="object" class="btn-primary" data-hotkey="q"
                                 attrs="{'invisible': [('subtype_is_log', '=', False)]}"/>
-                        <button string="Cancel" class="btn-secondary" special="cancel" data-hotkey="z" />
+                        <button name="cancel_composer" string="Cancel" class="btn-secondary" special="cancel" data-hotkey="z"/>
```
Comment (tde-banana-odoo) [inline addons/mail/wizard/mail_compose_message_views.xml:82]: Why this change ? All wizards of Odoo anyway when you cancel them ... why would we want another behavior here ? We globally never do that, as it adds unnecessary clicks to the user. You know what you do when you click on "cancel". Did you speak with @LNAodoo for this one, did he agree with adding clicks ?

Comment (alexkuhn) [reply addons/mail/wizard/mail_compose_message_views.xml:82]: Hey, this task is essentially a way to prevent accidental closing of full composer.

This is still in RD Dev. The code is matching the specs, which cover pressing ESC and also the buttons (X and Cancel).
We can ask our PR @paju-odoo for not considering X/Cancel buttons.

Comment (tde-banana-odoo) [reply addons/mail/wizard/mail_compose_message_views.xml:82]: I can read the code, thanks ^^

What I say is that it makes no sense, as the button tells you what is about to happen. Note that you may rename "Cancel" to "Discard" if it seems clearer. But we generally fight to remove useless clicks, and it seems we are adding clicks where it is not necessary, especially we don't do that anywhere else.

File: addons/mail/static/src/core/web/dialog_patch.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    /**
+     * @override
+     */
+    setup() {
+        this._super();
+        useHotkey("escape", () => {
+            if (this.data.model !== "mail.compose.message") {
+                this.data.close();
+            }
+        });
```
Comment (alexkuhn) [inline addons/mail/static/src/core/web/dialog_patch.js:17]: Instead of overshadowing `useHotkey` for all dialogs, better override an `onEscape` method.
Just to avoid duplicating the `this.data.close()`

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

File: addons/web/static/src/core/dialog/dialog.js [hunk 3]
```diff
@@ -74,6 +72,10 @@ export class Dialog extends Component {
         this.position.left = 0;
         this.position.top = 0;
     }
+
+    close() {
+        this.data.close();
+    }
```
Comment (alexkuhn) [inline addons/web/static/src/core/dialog/dialog.js:78]: I prefer naming it `onEscape`, just so we know in the override that it customises the behaviour of pressing `Esc` on keyboard.

Override on `close()` gives the wrong impression that we cannot close the dialog of `mail.compose.message` at all, including with buttons.

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise
